tests/static-constructor.c

```diff
@@ -0,0 +1,10 @@
+#include "gtest_like_c.h"
+
```

Comment:
I would keep only static-constructor.cpp.

Reply:
Non-static instance removed.

Reply:
I think static-constructor.c can be removed.

Reply:
Without the code in \`static-constructor.c\` the tests would not be included in the test suite, i.e. executed during the workflow run. Is that what you mean?

Reply:
```
TEST(Aarch64MinGW, StaticConstructorTest)
{
    ASSERT_TRUE(static_construtor());
}
```
This code is present in cpp. Is it not enough?

Reply:
That code is compiled in with real GTest, I made it consistent e.g. with [mingw-woarm64-build/tests/sscanf-double.cpp](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/tests/sscanf-double.cpp) even if the real GTest is currently not used. The C implementation of GTest cannot be compiled with C++ compiler.

Reply:
Even after placing TEST(Aarch64MinGW, StaticConstructorTest) in extern "C"?

Reply:
Yes, that works. Thank you for the suggestion. I was also wrong with the assumption that the C++ file were including/using the real GTest implementation.